### PR TITLE
Add java runtime optimizations

### PR DESCRIPTION
**Pull request by ShishirH**
Signed-off-by: Shishir Halaharvi <shihala1@in.ibm.com>

### CODE REVIEW

**Comment by chandrams** [comment]
Tested for runtime recommendations with this image shishirh/kruize:runtime, there is an issue with kruize generating runtime parameters for a newly deployed petclinic app. Code needs to be fixed, discussed this with @ShishirH

**Comment by ShishirH** [comment]
@dinogun Added the documentation in the latest commit. Please let me know if you think the changes are OK so that I can clean up the commit tree to a single commit.

File: src/main/java/com/kruize/service/RecommendationsService.java [hunk 1]
```diff
@@ -153,11 +160,64 @@ private JsonObject getResourceJson(ApplicationRecommendationsImpl applicationRec
...
+        JavaRecommendations javaRecommendations = JavaApplicationMetricsImpl.javaApplicationInfoMap
+                .get(labelName)
+                .getJavaRecommendations();
+
+        double heapRecommendations = javaRecommendations.getHeapRecommendation();
+
+        String gcPolicyRecommendation = javaRecommendations.getGcPolicy();
+
+        String percentage = precisionTwo.format((MathUtil.bytesToMB(heapRecommendations * 100)
+                /  applicationRecommendations.getRssLimits(application)));
```
Comment (dinogun) [inline src/main/java/com/kruize/service/RecommendationsService.java:205]: Isnt the getRssLimits in bytes as well and needs to be converted to MB ? If not, I would suggest that we need to be consistent and have all values sored either in MB or in bytes.

File: src/main/java/com/kruize/service/RecommendationsService.java [hunk 2]
```diff
@@ -153,11 +160,64 @@ private JsonObject getResourceJson(ApplicationRecommendationsImpl applicationRec
...
+    private JsonObject getRuntimeOptions(ApplicationRecommendationsImpl applicationRecommendations, String application) throws NoSuchApplicationException, NullPointerException
+    {
+        if (applicationRecommendations.getRuntime(application).equals("java"))
+        {
+            return getJavaOptions(applicationRecommendations, application);
+        }
+        return null;
+    }
+
+    private JsonObject getJavaOptions(ApplicationRecommendationsImpl applicationRecommendations, String application) throws NoSuchApplicationException, NullPointerException
```
Comment (dinogun) [inline src/main/java/com/kruize/service/RecommendationsService.java:188]: Please add function documentation javadoc

File: src/main/java/com/kruize/service/RecommendationsService.java (hunk 1 above)
Comment (ShishirH) [reply src/main/java/com/kruize/service/RecommendationsService.java:205]: Right now all the instances have the values in bytes, and all the applications have metrics in MB, so getRssLimits will be in MB.

Should I change all the instance recommendations, including rssRequests and rssLimits and the java recommendations to MB?

**Pull request merged**